pkg/test/utils/kubernetes.go

```diff
@@ -384,3 +386,51 @@ func SetAnnotation(obj runtime.Object, key, value string) runtime.Object {
 
 	return obj
 }
+
+// GetAPIResource returns the APIResource object for a specific GroupVersionKind.
+func GetAPIResource(dClient discovery.DiscoveryInterface, gvk schema.GroupVersionKind) (metav1.APIResource, error) {
+	resourceTypes, err := dClient.ServerResourcesForGroupVersion(gvk.GroupVersion().String())
+	if err != nil {
+		return metav1.APIResource{}, err
+	}
+
+	for _, resource := range resourceTypes.APIResources {
+		if !strings.EqualFold(resource.Kind, gvk.Kind) {
+			continue
+		}
+
+		return resource, nil
+	}
+
+	return metav1.APIResource{}, fmt.Errorf("resource type not found")
+}
+
+// WaitForCRDs waits for the provided CRD types to be available in the Kubernetes API.
+func WaitForCRDs(dClient discovery.DiscoveryInterface, crds []runtime.Object) error {
+	waitingFor := []schema.GroupVersionKind{}
+
+	for _, crdObj := range crds {
+		crd, ok := crdObj.(*apiextensions.CustomResourceDefinition)
+		if !ok {
+			continue
+		}
+
+		waitingFor = append(waitingFor, schema.GroupVersionKind{
+			Group:   crd.Spec.Group,
+			Version: crd.Spec.Version,
+			Kind:    crd.Spec.Names.Singular,
+		})
+	}
+
+	return wait.PollImmediate(100*time.Millisecond, 10*time.Second, func() (done bool, err error) {
+		allFound := true
+		for _, resource := range waitingFor {
+			_, err := GetAPIResource(dClient, resource)
+			if err != nil {
+				allFound = false
```

Comment:
why not return from here? ideally including the error

Reply:
I will make it return, but returning the error would cause it to stop retrying:

>  PollImmediate tries a condition func until it returns true, an error, or the timeout is reached.